airflow/contrib/operators/ecs_operator.py

```diff
@@ -1,30 +0,0 @@
-#
```

Comment:
This file can be removed only in Airflow 3 so we can not remove it yet.

Reply:
But it's so olllllldddddddd.    Git Blame shows that it was deprecated at least 2 years ago.  If it needs to stay, I'll add it back.

Reply:
When I revert this, what is the best way to handle it?   Should I import all of the new ECS Operators fro this PR and add them to `__all__`, or only update the names of the ones which were already included in there?

Reply:
Regardless contrib files cant not be remove. In this specific case we must be backward compatible till Airflow 3

As for your question - in this file you need to support only classes that existed. We dont add new functionality

Reply:
I have an idea for the future. Should we move alll the deprecated/contrib operators from 1.10 out to a separate "apache-contrib" package (when we split out providers). 

That will allow to get rid of the contrib, and get rid of many 1.10 deprecations out of airlfow main repo at least (it can still be a dependency of Airflow package so that it is installed automatically when airflow is installed). @ashb  @kaxil WDYT?

Reply:
> As for your question - in this file you need to support only classes that existed. We dont add new functionality

I thought I understood this, but the CI is proving me wrong.  Before this CR, contrib looked like this:
```
from airflow.providers.amazon.aws.operators.ecs import ECSOperator, ECSProtocol

__all__ = ["ECSOperator", "ECSProtocol"]

warnings.warn(
    "This module is deprecated. Please use `airflow.providers.amazon.aws.operators.ecs`.",
    DeprecationWarning,
    stacklevel=2,
)
```
and `deprecated_classes.py` had this:
```
(
    'airflow.providers.amazon.aws.operators.ecs.ECSOperator',
    'airflow.contrib.operators.ecs_operator.ECSOperator',
),
```
ECSOperator and ECSProtocol were deprecated some time ago in favor of the names EcsOperator and EcsProtocol, so it was importing classes which were deprecated.  Do those deprecated classes still have to stay now as well?  It has been a full major release of the provider package so I wanted to start removing those old methods.  In this PR I am now renaming  EcsOperator to EcsRunTaskOperator, and EcsProtocol has been deprecated and moved into *.hooks.ecs.EcsProtocol

Reply:
Well. They shoudl be. But this is not a must actually.

Contrib were part of Airflow 1.10. When we moved to 2, we vowed to keep MOSTLY compatibility  with 1.10. I.e. people who still used 1.10 would be able to continue using them. There were a number of exceptions and back then we had "backport" providers so people could simply move to use backports "airfow.providers.*" but we stoppped (as planned) releasing those, so the "late-comers" will have to deal with conesequences ot potentially broken compatibility if they still use contrib.

Since then many of those Contrib operators and hooks were modified and they are not even backwards-compatible with the original Contrib ones. 

So technically speaking, if an operator is anyhow backwards incompatible we do not have to keep it in contrib. But if it largely is, it makes it easier for user to have smooth migration.

Reply:
@eladkal - @potiuk mentioned wanting to get a provider package release out and Amazon would be getting a major release, I'm hoping to get this in on that but I'm not really sure what needs to be done here now.  Any help would be appreciated.

Reply:
It will take me a few days to get to this

Reply:
I can take this one. But you need to rebase @ferruzzi - this one has problems from few days back when we fixed Flask 2.2 compatibility:

![image](https://user-images.githubusercontent.com/595491/183124188-5642e25a-1050-469e-9717-cbd014bb6272.png)

Reply:
ACK.   Will rebase and do a force-push.  Thanks to both of you for the help.

Reply:
Done.

Reply:
I was a bit confused on how that chain of depredations should play out, but now with https://github.com/apache/airflow/pull/25543 I'm doubly confused on how the two will interact.